### ISSUE TITLE
Setup has detected that setup is currently running

### ISSUE THREAD

**Issue by chrmarti** [Microsoft/vscode#62476]
Issue Type: <b>Bug</b>

Unexpected since I did not run any setup myself. Shortly before that the (1) badge showed up on the gear icon.

![image](https://user-images.githubusercontent.com/9205389/47920110-8236b080-deb1-11e8-93d5-ab2a3839552c.png)

VS Code version: Code - Insiders 1.29.0-insider (220bff273eed39013f24b6d26fc17f2a742ce10f, 2018-11-02T06:18:20.695Z)
OS version: Windows_NT x64 10.0.17134

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-4770HQ CPU @ 2.20GHz (4 x 2195)|
|GPU Status|2d_canvas: enabled<br>checker_imaging: disabled_off<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>rasterization: unavailable_software<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Memory (System)|5.98GB (2.91GB free)|
|Process Argv|--inspect-extensions=51039|
|Screen Reader|no|
|VM|100%|

</details><details><summary>Extensions (15)</summary>

Extension|Author (truncated)|Version
---|---|---
remotehub|eam|0.2.0
EditorConfig|Edi|0.12.5
vscode-azureappservice|ms-|0.9.1
vscode-azurefunctions|ms-|0.12.0
vscode-azurestorage|ms-|0.4.2
vscode-cosmosdb|ms-|0.8.0
azure-account|ms-|0.4.3
azurecli|ms-|0.4.2
csharp|ms-|1.17.0
github-issues-prs|ms-|0.9.1
vscode-node-azure-pack|ms-|0.0.4
azurerm-vscode-tools|msa|0.4.2
debugger-for-chrome|msj|4.10.2
vscode-docker|Pet|0.3.1
vscode-open-in-github|sys|1.7.0

</details>

**Comment by msoutopico** [comment]
This also happens to me. There's only one instance running.

**Comment by misode** [comment]
This is happening regularly for me. Definitely an annoyance. Any updates?

**Comment by lando-securus** [comment]
![vs code setup error  need update](https://user-images.githubusercontent.com/52209841/68617693-60411800-048d-11ea-914d-e34abf4a97dc.PNG)

Seems to have solved it for me. I had two windows open and the second one was waiting on a restart.

**Comment by NicholasM-PSE** [comment]
I have the same issue message but the problem comes when running VSCode under two user accounts simultaneously in the same Windows session. I.e. user thisisme  launches vscode and then RunAs user thisismeadmin (separate AD accounts) a second vscode instance the second one will pop this message up every few minutes. If either instance is closed the issue goes away. -- is this a separate issue that the one discussed here?

**Comment by Siccity** [comment]
Same problem

**Comment by connerf** [comment]
Same issue

**Comment by AubreyR** [comment]
Same issue

**Comment by JasonLiebgott** [comment]
Same issue.

**Comment by ptr727** [comment]
Also happens when opening very large 3.5GB JSON file, memory already set at 8192MB, open file, editor says restart with memory set to 8192, ok, wait, popup that says setup already running.

**Comment by RoyalXVII** [comment]
![image](https://user-images.githubusercontent.com/51414638/117043816-ffa33c00-ad0d-11eb-9e71-ca12cb04854b.png)

**Comment by varunkumarnr** [comment]
I copied the "Code.exe" file from where my program was installed and pasted it to the temp map: C:\Users\YOUR_USERNAME\AppData\Local\Programs\Microsoft VS Code I reran the update and it fixed the problem for me.

**Comment by Temepest74** [comment]
well, this is lost between 5k+ issues. Maybe this needs to be brought back to life.

**Comment by DanaFo** [comment]
I have the same thing. I tried uninstalling code and reinstalling it but during install it gives me Setup has detected setup is running and I feel like I f'd myself.

**Comment by arielsashcov** [comment]
Getting the same error - Possibly because I manually moved Code.exe to another directory

**Comment by Henry4Xperit** [comment]
I also experience this issue:
![image](https://user-images.githubusercontent.com/86042318/143010152-4d47bb99-a187-4e3b-a663-84de1defc097.png)

**Comment by syborg64** [comment]
fix this !!!

**Comment by stephenchildree** [comment]
I'm also running into this issue. It pops up every 5 minutes or so and I can't figure out how to fix it.

**Comment by brmonaghan** [comment]
same here
![image](https://user-images.githubusercontent.com/71337477/168159933-1d3998b3-888b-49fc-ad24-0349a8dfb176.png)

**Comment by cypher437** [comment]
same

**Comment by chromial** [comment]
perhaps 'Extension host' is the culprit. I had to click away the 'Setup has detected that Setup is currently running.' message 3 times.
<img width="357" alt="image" src="https://user-images.githubusercontent.com/1315665/173074079-3ac662eb-5d41-4893-b9bd-d73b7a84554c.png">

**Comment by mattwojo** [comment]
Same issue :(

**Comment by SummerWang00** [comment]
I have the same issue too at VS code start up. Sometimes it doesn't happen immediately after startup, but a few minutes after the startup, the same window pops up. Clicking cancel makes it go away but it would come back after one hour or so. I am not sure what is causing this. I'm using the newest version of VS code as for now.

**Comment by NicholasM-PSE** [comment]
Suggestions to fix? 
I would love to see this either fail silently or perhaps just become a notification rather than a popup that has to be dismissed. Additionally if setup does find that setup is running, could it maybe wait a little longer before trying setup again?

**Comment by oskarols** [comment]
Additionally, surely there would be some way of indicating that this is related to VSC. I got randomly prompted for this. I had no idea what it was.

**Comment by kmilo9999** [comment]
![image](https://user-images.githubusercontent.com/16551165/183506198-d19e7e9e-fc74-4e66-b025-b75b08f76db7.png)
same issue here.

**Comment by Superhands89** [comment]
same issue !!!

**Comment by Hiren547** [comment]
Guys, Don't worry about. Try this type of path : Users\yourfoldername\AppData\Local\Programs\Microsoft VS Code

This path will show after you create your VS shortcode icon and then right click the icon and go to the file path and you are able to see this type of path.  it's working for me:)

**Comment by pShota** [comment]
I can same popup very frequently in Win 10 now, very annoying.
@Hiren547  I cannot fully understand what its means (no offense)
do you mean create a new short cut in desktop??

**Comment by justingolden21** [comment]
I've been using vs code for years just fine without seeing this. The past few days, this has popped up every hour. I have one instance of vs code open. I'm not doing anything wacky. I haven't updated vs code recently. Couldn't even figure out what program was sending me these stupid prompts until I checked task manager. Is this issue really over 4 years old?

**Comment by justingolden21** [comment]
Lis of extensions:

![image](https://user-images.githubusercontent.com/30274440/218567977-6f82b71c-b478-4381-911d-ba54c4355497.png)

**Comment by thistlillo** [comment]
+1 here since this morning. VSC was running mostly fine yesterday, then this morning this issue showed up. I did not do anything during the night.

**Comment by justinswork** [comment]
Same issue. Hitting "OK" just makes the dialog pop back up again. Super frustrating/annoying

**Comment by kurapov-peter** [comment]
Same issue.

**Comment by stevenbyerly** [comment]
Getting this as well, every day it pops up even with no VSCode instances open.

**Comment by skeddles** [comment]
this happens to me all the time. cant believe this bug is 5 years old and still not fixed.

**Comment by weiyudaixing** [comment]
still getting it and come to this place...

**Comment by Yu-zk** [comment]
Same issue here

**Comment by skyiky** [comment]
5 years no fix LMAO

**Comment by ChillHumanoid** [comment]
Mine just started doing this today. You can click 'cancel' on the pop up and it will go away... temporarily. Definitely an annoying bug, especially finding out this is such an old issue.

**Comment by justingolden21** [comment]
Every time I power off my PC I get this. FIVE YEARS. And I'm not the only one. Wild...

Anybody have any ideas on the cause or the solution?

**Comment by dc6c** [comment]
For me, there were some files in, IIRC, `%HOME%/AppData/Roaming/Code` that were not writeable. Once I accidentally it found, I just reset the permissions and everything worked again.

**Comment by skeddles** [comment]
I think I fixed it by deleting all old instances of VS code on my hard drives

**Comment by Jon4242** [comment]
In my case it was because there was a process called CodeSetupStable.exe running that was hung somehow and never exited even when VSCode was terminated. Killing it in task manager fixed it. Presumably rebooting might have fixed it too but it is possible that CodeSetupStable.exe might restart if it thought it hadn't finished yet?

**Comment by brianlmerritt** [comment]
If it's any help to Microsoft or anyone else, def happening on my Windows 11 desktop with two VSCode windows open, one running Python code  launched with a Conda virtual environment and one running C++ launched via PowerShell for VS 2022.  I could happily run 2 VSCode windows launched from the TaskBar.

**Comment by Murtatrxx** [comment]
We definitely need attention!

**Comment by emptyopen** [comment]
+1

**Comment by Gruntfuggly** [comment]
Just had this out of nowhere. Had to google it to find that it was VSCode.

Could somebody at least just change the ridiculous message?

**Comment by MoistGoolem** [comment]
Just happened here as well. Please change the message or handle it differently 👍

**Comment by CocoCodeColada** [comment]
Hi I am 95% sure that i have this problem for the following reason:

My VS Code is super fresh. I have barely anything installed, no files, pretty clean (except that I couldn't find the installed MS Python Interpreter in my files). So me being quite the newbie tried to set the Python Interpreter manually and in the process of trying things out i kind of set the VS Code Installer exe as the Interpreter. i know. please don't judge. I got the preobem immediately after and deleting vs code and reinstalling didnt solve the problem. i also still cant find the interpreter.

**Comment by aworld1** [comment]
Hey all, for anyone who is encountering this error, we were able to solve it in a few steps.

1. Go to your downloads, delete every instance of the VS Code setup exe
2. Empty your recycling bin to completely delete these files
3. End every instance of VS code and setups in the task manager
4. Restart your computer
5. Launch VS Code
6. Voila

**Comment by xmaxrayx** [comment]
had it today , most funny error message i have seen , thx devs.

**Comment by lucasrossell** [comment]
still happening in 2024. what an embarrassing bug for such a major Microsoft product

**Comment by miles-manulife** [comment]
This is happening for me in December 2024 - normally I'm a Mac user and I've never seen this before, so it must be part of my penance for using Windows on my current project

**Comment by hingbong** [comment]
Got this too

**Comment by rndninja** [comment]
I got this error as well. Still happening in 2025 May.

**Comment by HelloWook** [comment]
I got this error as well.

**Comment by vxsourcerer** [comment]
Got this! 
June 25 and it is still happening.

**Comment by GemPG** [comment]
I have started seeing this only in the last couple of weeks. Possibly after having finally clicked on an "upgrade" notification.
Perhaps related to having multiple projects open, but I often do/did use multiple projects.

**Comment by TheLastAirbenderr** [comment]
I have the same problem - absolutely ridiculous

**Comment by justinswork** [comment]
Someday a higher-up manager at Microsoft will experience how stupid and annoying this is and then it'll get fixed immediately 😅

**Comment by PURASHADOW** [comment]
Just restart

**Comment by legopitstop** [comment]
I found a workaround.

1. Open task manager once you're on "Installing Update..." and have given it some time to update.
2. In the Processes tab, search for "PowerShell"
3. Kill that process, and it will successfully install.

No more annoying Setup window.

**Comment by jtc10005** [comment]
I've been getting this for months now...

**Comment by jtc10005** [comment]
> I found a workaround.
> 
> 1. Open task manager once you're on "Installing Update..." and have given it some time to update.
> 2. In the Processes tab, search for "PowerShell"
> 3. Kill that process, and it will successfully install.
> 
> No more annoying Setup window.

well.... I'll be... this worked for me....  Thank you @legopitstop

**Comment by dmitrivMS** [comment]
@legopitstop Do you happen to remember if you had 2 or more instances of VS Code running?

**Comment by legopitstop** [comment]
The "Setup" dialog only shows up after opening a new instance.

**Comment by alexhass** [comment]
This happens in 1.22.1, too. Last 6-8 updates in a row - all the same bug.

@dmitrivMS This happens with one instances of vscode.

**Comment by dmitrivMS** [comment]
Verification steps:
- Install an older build of Insiders (from builds page)
- Start VS Code
- Start several mode instances (e.g. from command-line "code ...")
- Wait for Update to kick in (you will see Update button show up in all windows)
- Make sure you don't see the error dialogs above
- Let it restart, verify update worked

**Comment by alexhass** [comment]
How about implementing a great MSI setup and leave the InnoSetup crap behind?

**Comment by dmitrivMS** [comment]
@alexhass I believe this one is tracking that ask: https://github.com/microsoft/vscode/issues/61793

**Comment by alexhass** [comment]
No. Not MSIX - MSI.

**Comment by alexhass** [comment]
@dmitrivMS  The installation / upgrade bug is still not fixed in 1.125.1. Please reopen. Logfile attached.

[Microsoft Visual Studio Code_1.125.1_Install.log](https://github.com/user-attachments/files/29142408/Microsoft.Visual.Studio.Code_1.125.1_Install.log)

**Comment by dmitrivMS** [comment]
@alexhass Did you hit it while upgrading from 1.123 or from a later version? The reason I ask, is the bug is on the version that's doing the install - so if you were running an unfixed version, it would still manifest.

**Comment by alexhass** [comment]
installed 1.125.1 over 1.122.1 without uninstalling 1.122.1.

but i'm running an upgrade by running setup exe from latest 1.125.1 version... this is done via intune. I'm not using the build-in update mechanism.

If I uninstall 1.122.1 and than install 1.125.1 it works. This is my current workaround.

**Comment by dmitrivMS** [comment]
The fix helps upgrade from 1.124 and above to newer versions, so it won't do much good for upgrading from 1.123 or below. If you hit it when upgrading e.g. from 1.125 to 1.126, please let me know.

**Comment by alexhass** [comment]
Just tried updating my private machine from 1.124 to 1.125.1 (without intune). In this specific case - two vscode's have been open and auto-closed by the setup. I guess there is no log as I ran the update with the update button in vscode. At least I have a screenshot of what happens (english "permission denied")

<img width="401" height="214" alt="Image" src="https://github.com/user-attachments/assets/1f71b298-34fd-4358-9137-4fbeda56832b" />

I think I'm not going to try this again to 50 developers with intune once 1.126 gets released.

This setup is proven to - not working properly. :-)

**Comment by dmitrivMS** [comment]
@alexhass Thank you for the info. That is a different error which I'd like to get to the bottom of as well, but please open a new issue on it.
The good news is you don't see "Setup has detected that Setup is currently running." any longer.

**Comment by alexhass** [comment]
No. It is the same error I see for months. If the message window is shown - the setup log stops. Setup sis stale and do not continue. If this happens during automated intune deployment - setup is stale and does not complete before you kill the setup process / restart the computer / logs off.
